purchase_request/models/purchase_order.py

```diff
@@ -194,8 +194,9 @@ def _prepare_request_message_data(
     def write(self, vals):
         #  it is done here instead of method _update_received_qty
         #  to make sure this work for services
-        prev_qty_received = self.qty_received
-        res = super(PurchaseOrderLine, self).write(vals)
-        if vals.get('qty_received', False):
-            self.update_service_allocations(prev_qty_received)
+        for line in self:
+            prev_qty_received = line.qty_received
+            res = super(PurchaseOrderLine, line).write(vals)
```

Comment:
I think the super() should be moved outside the loop and can be called only once, over self instead of each line.
As @hveficent  said, if you provide a use case or the steps to reproduce the issue, we can have a better look at this issue.

Reply:
Will this works?

```
    @api.multi
    def write(self, vals):
        #  it is done here instead of method _update_received_qty
        #  to make sure this work for services
        res = super(PurchaseOrderLine, self).write(vals)
        if vals.get('qty_received', False):
            for line in self:
                prev_qty_received = line.qty_received
                line.update_service_allocations(prev_qty_received)
        return res
```

Reply:
Much better, @CasVissers apply this.